### ISSUE TITLE
gosec: latest version (v2.17.0) returning false positives

### ISSUE THREAD

**Issue by Smelton01** [golangci/golangci-lint#4037]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've read the typecheck section of the FAQ (https://golangci-lint.run/usage/faq/#why-do-you-have-typecheck-errors).
- [X] Yes, I've tried with the standalone linter if available (e.g., gocritic, go vet, etc.). (https://golangci-lint.run/usage/linters/)

### Description of the problem

The latest [release v1.54.2](https://github.com/golangci/golangci-lint/releases/tag/v1.54.2) of golangci-lint includes an updated version of of [gosec v2.17.0](https://github.com/securego/gosec/releases/tag/v2.17.0) with changes which introduced a number of false positives.

The changes in question were introduced in the following PRs:
- https://github.com/securego/gosec/pull/971
  - This seems not properly tested because there's already an [issue](https://github.com/securego/gosec/issues/1001) with multiple people complaining about false positives.
-  https://github.com/securego/gosec/pull/1003
   - Another overlooked behavior resulting in false positive as reported in https://github.com/securego/gosec/issues/966#issuecomment-1683619581

Since this is a very popular linter trusted by many, a release should be made reverting the linter to a stable version until the above issues are addressed.

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version 1.54.2 built with go1.21.0 from 411e0bb on 2023-08-21T11:04:00Z
```

</details>

### Configuration

<details>

```console
default-config
```

</details>

### Go environment

<details>

```console
$ go version 
go version go1.21.0 darwin/arm64
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
...G101: Potential hardcoded credentials (gosec)
        SQSQueueURL  = "workertest-sqs-queue-url"
```

</details>

### A minimal reproducible example or link to a public repository

<details>

```go
// add your code here
```

</details>

### Validation

- [X] Yes, I've included all information above (version, config, etc.).

**Comment by vitaly-eureka-security** [comment]
The relevant issue in gosec repo: https://github.com/securego/gosec/issues/1001

**Comment by ldez** [comment]
I'm aware but there is no release :shrug:
